Handle continous legs

terencebls opened the pull request.

terencebls committed:
handle continous leg for walk

r3to committed:
Merge branch 'main' into handle-continous-legs

# Conflicts:
#	Sources/OJP/Models/OJPv2+Trip.swift

r3to committed:
Some Progress on ContinousLegs (blocked by https://github.com/openTdataCH/ojp-sdk/issues/70)

r3to committed:
Fix issue with darkmode

r3to committed:
service property on TimedLeg is actually of type DatedJourney

r3to commented:
see https://github.com/openTdataCH/ojp-sdk/issues/81 for a discussion about the required elements on ContinuousService

r3to committed:
Update ContinuousService to support both possible service elements

r3to committed:
Run swift format

r3to committed:
Remove print statements in tests

r3to committed:
That's probably the fix for the failure, but I have no idea, why this only fails on github and not on our machines

r3to committed:
Add debug dump for unit test on github

r3to committed:
Use decodeIfPresent to avoid Swift-Error Breakpoints

r3to committed:
Clean up and re-enable tests

r3to committed:
merge main into handle-continouou-legs

commit 3a8416c049e3de872394ba7a0ae6f6d98b247fc6
Merge: 9d8fa50e 5f50a979
Author: terencebls <85493933+terencebls@users.noreply.github.com>
Date:   Wed Aug 14 16:18:37 2024 +0200

    Merge pull request #65 from openTdataCH/feature/situations

    Feature/situations

commit 5f50a9799e0fa7e981e8c72042bfc0fff3efd0b1
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Tue Aug 13 11:46:28 2024 +0200

    make RealtimeData as enum

commit 4182cb5104e0598d0a4cc7eb3a979f455acc936e
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Tue Aug 13 11:37:46 2024 +0200

    Add more situation example files xml

commit 8d88c19e02593296e2f8c374adaf3e68f44faf11
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Tue Aug 13 11:36:54 2024 +0200

    add useRealtimeData param

commit 141b41d1bcfdf68557c34641cab38c9f6a0fddba
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Mon Aug 12 17:53:11 2024 +0200

    Add DetailView for PTSituations

commit 78affb10f6c82eb7f7ca463aedf14b1a3d844e1f
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Mon Aug 12 17:48:32 2024 +0200

    Finally getting the Situations to parse correctly (Situations aren't a choice type....) and clean up the test

commit a368ea0f4616f42a347991cdaed4f8b04e4f99d1
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Mon Aug 12 11:59:49 2024 +0200

    Cosmetics

commit ac85f421d8bee188e1a4e7e54fe42ebef1b436f3
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Mon Aug 12 10:41:59 2024 +0200

    Move convenience extensions from SampleApp to SDK

commit 6d89ba5d4d3ab0c4025f0b67c6cd041a99206e63
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Mon Aug 12 10:35:51 2024 +0200

    Add test case with multiple situations

commit 4558eda8d7401406fa629ceec0dcdaf4d3d5b949
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 15:19:22 2024 +0200

    Make mapping a little bit more easier to read (hopefully)

commit d7873e393d441080ee9793af871aabef65fc4d36
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 15:18:44 2024 +0200

    Switch to preview content with situations

commit 5e65b54212b608dfac1498a6c768b7b8da2d2e01
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 14:55:25 2024 +0200

    Use Identifiable instead of Hashable

commit 8c10529266d237361635fa15e52644eb4928672c
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 14:33:25 2024 +0200

    Enhance AsyncView and PreviewMocker for cases, where loading fails

commit 14ade7103f62ad79903a4df9a9a8f53c04c67ca8
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 14:07:13 2024 +0200

    Remove unused Test XML

commit d0cf37628a376828ffda4e5271be1c0515b27891
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 13:53:11 2024 +0200

    Run Swift Format

commit 8c07c685fb56ac347575f27d7c18559fbee615a6
Author: Reto Lehnherr <reto.lehnherr@ti8m.ch>
Date:   Fri Aug 9 13:50:35 2024 +0200

    rename Service to DatedJourney

commit 5ad9c9507fd8bd3ecfbc03c5fa09507afa8fcd6d
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Fri Aug 9 11:41:58 2024 +0200

    show PTSituations on test app

commit 5ea9195768bb489cf1b292a5dab405c54a0e34f8
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Thu Aug 8 16:51:30 2024 +0200

    add situations on trip

commit 7e0b0be76e1aeb9261640bf402533039fb6f459c
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Thu Aug 8 15:50:15 2024 +0200

    better handle the hack for empty array (with custom init)

commit 9c15bdc4352ba4ffb588f9d036982f84ad275f67
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Thu Aug 8 15:39:52 2024 +0200

    avoid parsing bug when one of the arrays is empty

commit 981f7d15165c61268b012e530da6681af932017c
Author: Terence Alberti <terence.alberti@bls.ch>
Date:   Thu Aug 8 14:52:47 2024 +0200

    add model for Situations

commit 9d8fa50e93ccec1b221648a59cf34814441e3e38
Author: Reto Lehnherr <reto.lehnherr@mac.com>
Date:   Tue Jul 30 15:11:40 2024 +0200

    56 decoder has issue with namespace handling when an element name can is in both namespaces (#58)

    * Start defining TripResonseContext

    * WIP finding a solution for namespace issue

    * Ignoring positional keys, fixing the tests

    * Add missing Sendable conformance

    * set await

    ---------

    Co-authored-by: Terence Alberti <terence.alberti@bls.ch>
